### ISSUE TITLE
Fix `selector-max-specificity` false positives with `ignoreSelectors` option for `of <selector>` syntax

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#6615]
### What steps are needed to reproduce the bug?

The following second CSS snippet (selector) reports a `selector-max-specificity` problem:

```css
/* pass */
foo:nth-child(even) { color: red; }

/* fail */
foo:nth-child(even of a) { color: red; }
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-max-specificity": ["0,1,0", { "ignoreSelectors": ["/foo/"] }]
  }
}
```

### How did you run Stylelint?

CLI

### Which version of Stylelint are you using?

`14.16.1`

### What did you expect to happen?

No problems to be reported by the `ignoreSelectors` option.

### What actually happened?

The following problem is reported:

```
5:1  Expected "foo:nth-child(even of a)" to have a specificity no more than "0,1,0"
```

See the [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcID0AqABABwIYGcdpSQB0A7AMwgnhIBcALAWjDoEsAbKAChgDcYSAlGmBpIbCACd4aCTCgBuNAF9SpVGjJZ2BYuUrV6TVh258SaCGTRYhIsZOmyFy0iAA0IHAE9aWAB4I4Hju4BDkLADmgcCkaGhEIBIArmwwOAnSMeZx8Z4wqWA0kgwAtv4MOBgwYCxkLDU0XhloANoJAAxuAIxu7QluwrmRJJIwAMr51UUS6QitCUgUEMQgALrKq7EuJEogSkA).

### Proposal to fix the bug

Relevant code:

- https://github.com/stylelint/stylelint/blob/b3af7d2bb8cce81bb7e51e81d980274bb4d14643/lib/rules/selector-max-specificity/__tests__/index.js#L463-L466
- https://github.com/stylelint/stylelint/blob/b3af7d2bb8cce81bb7e51e81d980274bb4d14643/lib/rules/selector-max-specificity/index.js#L129-L131

See also #6140

**Comment by ybiquitous** [comment]
I'm dealing with this issue by parsing the "of selector" syntax (`:nth-child(An+B of S)`) ([spec](https://drafts.csswg.org/selectors/#the-nth-child-pseudo)).

**Comment by ybiquitous** [comment]
I've clarified the issue title according to the MDN wording (`of <selector>` syntax).
Ref: https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child
<img width="176" alt="image" src="https://github.com/stylelint/stylelint/assets/473530/9a8c76e7-741e-43ff-ac8e-3648ed805bcc">
